homeassistant/components/zone/config_flow.py

```diff
@@ -0,0 +1,69 @@
+"""Config flow to configure zone component."""
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.const import (
+    CONF_NAME, CONF_LATITUDE, CONF_LONGITUDE, CONF_ICON, CONF_RADIUS)
+from homeassistant.core import callback
+from homeassistant.util import slugify
+
+from .const import CONF_PASSIVE, DOMAIN
+
+
+@callback
+def configured_zones(hass):
+    """Return a set of the configured hosts."""
+    return set((slugify(entry.data[CONF_NAME])) for
+               entry in hass.config_entries.async_entries(DOMAIN))
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class ZoneFlowHandler(data_entry_flow.FlowHandler):
+    """Zone config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize zone configuration flow."""
+        pass
+
+    async def async_step_init(self, user_input=None):
+        """Handle a flow start."""
+        errors = {}
+
+        if user_input is not None:
+            if slugify(user_input[CONF_NAME]) not in \
+               configured_zones(self.hass):
+                return self.async_create_entry(
+                    title='Zone ' + user_input[CONF_NAME],
+                    data=user_input,
+                )
+            errors['base'] = 'name_exists'
+
+        return self.async_show_form(
+            step_id='init',
+            data_schema=vol.Schema({
+                vol.Required(CONF_NAME): str,
+                vol.Required(CONF_LATITUDE): cv.latitude,
```

Comment:
We will need to update voluptuous-serialize to allow registering custom types.

Reply:
It works though

Reply:
it works because of how we implemented cv.latitude, it's just a range. nice.